code/game/machinery/doors/firedoor.dm

```diff
@@ -290,11 +294,12 @@
 	obj_flags |= EMAGGED
 	INVOKE_ASYNC(src, .proc/open)
 
-/obj/machinery/door/firedoor/Bumped(atom/movable/AM)
+/obj/machinery/door/firedoor/Bumped(atom/movable/movable)
 	if(panel_open || operating)
 		return
 	if(!density)
 		return ..()
+	SEND_SIGNAL(src, COMSIG_ATOM_BUMPED, movable)
```

Comment:
this is really horrible you know

Reply:
I don't understand the significance of it, to tell you the truth. It doesn't return parent by default, so how are you supposed to get all the way up the chain to where it calls BUMPED at the atom level? Do I have to make it call parent and then fix whatever unintended functionality arises from that?

Reply:
![image](https://user-images.githubusercontent.com/58055496/156876936-eefa7afd-14bf-4d7d-9b75-88b0e81bc741.png)

Reply:
I've gone through that and from what I can tell by calling parent the only issue that arises with firelocks is throwing your ID to open them which I've fixed in the code I'm about to test then push.